components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -0,0 +1,653 @@
+     * Content-Type: application/x-www-form-urlencoded
+     * <p/>
+     * grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Ajwt-bearer
+     * &assertion=eyJhbGciOiJFUzI1NiJ9.
+     * eyJpc3Mi[...omitted for brevity...].
+     *
+     * @param tokReqMsgCtx Token message request context
+     * @return true if validation is successful, false otherwise
+     * @throws org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception
+     */
+    @Override
+    public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        super.validateGrant(tokReqMsgCtx);
+
+        SignedJWT signedJWT;
+        IdentityProvider identityProvider = null;
+        String tokenEndPointAlias = null;
+
+        // get the tenant domain from the token request message context
+        tenantDomain = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getTenantDomain();
+
+        if (tenantDomain == null || "".equals(tenantDomain)) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+
+
+        // Try to get the assertion from the Oauth2AccessTokenReqDTO assertion attribute
+        String assertion = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getAssertion();
+
+
+        // Since the assertion was not found in the assertion variable of the Oauth2AccessTokenReqDTO
+        // we try to find it in the request parameters
+        if (assertion == null) {
+            RequestParameter[] params = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getRequestParameters();
+
+            for (RequestParameter param : params) {
+                if (param.getKey().equals(OAUTH_JWT_ASSERTION)) {
+                    // if there is a param called assertion
+                    assertion = param.getValue()[0];
+                    if (log.isDebugEnabled()){
+                        log.debug("JWT assertion found in request parameters: " + assertion);
+                    }
+                    break;
+                }
+            }
+
+            // check whether a non empty assertion string was acquired
+            if (assertion == null || StringUtils.isEmpty(assertion)) {
+                log.error("No Valid Assertion was found for " + OAUTH_JWT_BEARER_GRANT_TYPE);
+                return false;
+            }
+        }
+
+        //logging the JSON Web Token(JWT) assertion
+        if (log.isDebugEnabled()) {
+            log.debug("Received JSON Web Token assertion: " + assertion);
+        }
+
+        try {
+            signedJWT = SignedJWT.parse(assertion);
+
+            // logging the parsed JWT
+            if (log.isDebugEnabled()) {
+                logJWT(signedJWT);
+            }
+
+        } catch (ParseException e) {
+            // there's some error in the JSON Web Token
+            log.error("Error parsing the JWT received " + e.getMessage(), e);
+            return false;
+        }
+
+        // get the claims from the parsed JWT
+        ReadOnlyJWTClaimsSet claimsSet;
```

Comment:
code is much clean if the required variables are declared at the top of method and istantiated as required.